tests/rptest/services/redpanda.py

```diff
@@ -832,6 +833,188 @@ def __init__(self,
+        start within a timeout period the service will fail to start. Thus this
+        function also acts as an implicit test that redpanda starts quickly.
+        """
+        pass
+
+    def stop_node(self, node, **kwargs):
+        """Halt service process(es) on this node."""
+        pass
+
+    def clean_node(self, node, **kwargs):
+        """Clean up persistent state on this node - e.g. service logs, configuration files etc."""
+        pass
+
+    def start(self,
+              nodes=None,
+              clean_nodes=True,
+              start_si=True,
+              parallel: bool = True,
+              expect_fail: bool = False,
+              auto_assign_node_id: bool = False,
+              omit_seeds_on_idx_one: bool = True,
+              node_config_overrides={}):
+        """
+        Start the service on all nodes.
+
+        By default, nodes are started in serial: this makes logs easier to
+        read and simplifies debugging.  For tests starting larger numbers of
+        nodes where serialized startup becomes annoying, pass parallel=True.
+
+        :param parallel: if true, run clean and start operations in parallel
+                         for the nodes being started.
+        :param expect_fail: if true, expect redpanda nodes to terminate shortly
+                            after starting.  Raise exception if they don't.
+        """
+        pass
+
+    def wait_until(self, fn, timeout_sec, backoff_sec, err_msg=None):
+        """
+        Cluster-aware variant of wait_until, which will fail out
+        early if a node dies.
+
+        This is useful for long waits, which would otherwise not notice
+        a test failure until the end of the timeout, even if redpanda
+        already crashed.
+        """
+        pass
+
+    def get_node_memory_mb(self):
+        pass
+
+    def get_node_cpu_count(self):
+        pass
+
+    def get_node_disk_free(self):
+        pass
+
+    def set_resource_settings(self, rs):
+        pass
+
+    def lsof_node(self, node: ClusterNode, filter: Optional[str] = None):
+        """
+        Get the list of open files for a running node
+
+        :param filter: If given, this is a grep regex that will filter the files we list
+
+        :return: yields strings
+        """
+        pass
+
+    def set_si_settings(self, si_settings: SISettings):
+        pass
+
+    def add_extra_rp_conf(self, conf):
+        pass
+
+    def storage(self, all_nodes: bool = False, sizes: bool = False):
+        """
+        :param all_nodes: if true, report on all nodes, otherwise only report
+                          on started nodes.
+
+        :returns: instances of ClusterStorage
+        """
+        pass
+
+    def set_extra_rp_conf(self, conf):
+        pass
+
+    def metrics_sample(
+        self,
+        sample_pattern,
+        nodes=None,
+        metrics_endpoint: MetricsEndpoint = MetricsEndpoint.METRICS,
+    ) -> Optional[MetricSamples]:
+        """
```

Comment:
Let's not duplicate docstrings, they'll get stale.  Either omit them on the base class or the derived class (not strongly opinionated on which)

Reply:
i'll remove them from the base class so the diff is smaller